in_toto/resolver/_resolver.py

```diff
@@ -262,3 +268,95 @@ def hash_artifacts(self, uris):
             os.chdir(original_cwd)
 
         return hashes
+
+
+class DirectoryResolver(Resolver):
+    """Directory resolver implementation.
+
+    Implementation of the following shell command for some directory URI:
+
+    find <URI> -type f | cut -c3- | LC_ALL=C sort | xargs -r sha256sum |
+        sha256sum | cut -f1 -d' '
+    """
+
+    SCHEME = "dir"
+
+    def __init__(
+        self,
+        exclude_patterns=None,
+        base_path=None,
+        follow_symlink_dirs=False,
+        lstrip_paths=None,
+    ):
+        if not lstrip_paths:
+            lstrip_paths = []
+        self._lstrip_paths = lstrip_paths
+
+        self.file_resolver = FileResolver(
+            exclude_patterns=exclude_patterns,
+            base_path=base_path,
+            follow_symlink_dirs=follow_symlink_dirs,
+            normalize_line_endings=True,
+        )
+
+    def _strip_scheme_prefix(self, path):
+        """Helper to strip file resolver scheme prefix from path."""
+
+        return path[len(self.SCHEME + ":") :]
+
+    def _mangle(self, path, existing_paths):
+        """Helper for path mangling."""
+
+        # Normalize slashes for cross-platform metadata consistency
+        path = path.replace("\\", "/")
+
+        # Left-strip names using configured path prefixes
+        for lstrip_path in self._lstrip_paths:
+            if path.startswith(lstrip_path):
+                path = path[len(lstrip_path) :]
+                break
+
+        # Fail if left-stripping above results in duplicates
+        if self._lstrip_paths and path in existing_paths:
+            raise PrefixError(
+                "Prefix selection has resulted in non unique dictionary key "
+                f"'{path}'"
+            )
+
+        # Prepend passed scheme prefix
+        path = self.SCHEME + ":" + path
+
+        return path
+
+    def _hash(self, exhaustive_hashes):
+        """Helper to correctly sort and hash every element."""
+
+        keys = list(exhaustive_hashes.keys())
+        locale.setlocale(locale.LC_ALL, "C")
+        keys.sort(key=cmp_to_key(locale.strcoll))
+
+        text_repr = "\n".join(
+            [f"{exhaustive_hashes[k][_HASH_ALGORITHM]}  {k}" for k in keys]
+        )
+        text_repr += "\n"  # trailing new line
+
+        digest_obj = digest(_HASH_ALGORITHM)
+        digest_obj.update(text_repr.encode("utf-8"))
+
+        return {_HASH_ALGORITHM: digest_obj.hexdigest()}
+
+    def hash_artifacts(self, uris):
+        hashes = {}
+
+        for path in uris:
+            path = self._strip_scheme_prefix(path)
+
+            exhaustive_hashes = self.file_resolver.hash_artifacts([path])
+            if len(exhaustive_hashes) == 0:
```

Comment:
Thy pythonic way would be...
```suggestion
            if not exhaustive_hashes:
```